client/my-sites/plugins/plugin-management-v2/index.tsx

```diff
@@ -1,20 +1,42 @@
-import { Button } from '@automattic/components';
 import classNames from 'classnames';
-import { useTranslate } from 'i18n-calypso';
-import { ReactElement, useEffect } from 'react';
-import ButtonGroup from 'calypso/components/button-group';
-import TextPlaceholder from 'calypso/jetpack-cloud/sections/partner-portal/text-placeholder';
+import { TranslateResult, useTranslate } from 'i18n-calypso';
+import { ReactElement, useEffect, useMemo } from 'react';
 import { useDispatch } from 'calypso/state';
 import { resetPluginStatuses } from 'calypso/state/plugins/installed/status/actions';
+import BulkActionsHeader from './bulk-actions-header';
 import PluginsList from './plugins-list';
-import UpdatePlugins from './update-plugins';
-import { pluginsEmptyMessage } from './utils/get-plugins-empty-message';
 import type { PluginComponentProps } from './types';
 import type { SiteDetails } from '@automattic/data-stores';
 import type { PluginFilter } from 'calypso/state/plugins/installed/types';
 
 import './style.scss';
 
+export const useEmptyMessage = (
+	searchTerm: string,
+	filter: PluginFilter
+): TranslateResult | undefined => {
+	const translate = useTranslate();
+
+	return useMemo( () => {
+		if ( searchTerm ) {
+			return translate( 'No results found. Please try refining your search.' );
+		}
+
+		const getMessage = (
+			{
+				all: ( translate ) => translate( 'No plugins found.' ),
+				active: ( translate ) => translate( 'No plugins are active.' ),
+				inactive: ( translate ) => translate( 'No plugins are inactive.' ),
+				updates: ( translate ) => translate( 'All plugins are up to date.' ),
```

Comment:
Why are we passing `translate` as an argument here?

Reply:
It's not strictly necessary, but it was my attempt at a performance optimization. By deferring the translation until we know for sure which message we need, we're saving ourselves 4 calls to `translate` each time this hook gets called.

More concretely:
- If we directly translate `all`, `active`, `inactive`, and `updates`, we have an object with translations for all four messages, regardless of the value of `filter`.
- If we write each message as a function that takes `translate` as an argument, nothing actually gets translated until the call to `getMessage?.( translate )` on line 36. And at that point, only the one message we need to display is translated.